restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/rcml/Parser.java

```diff
@@ -152,6 +162,235 @@ private Tag next() throws LimitExceededException{
         return null;
     }
 
+	///////////////////////////////////////////// new class -- validate
+    private boolean validation(Tag tag) {
+    	boolean shouldContinue=true;
+    	List<Attribute> mapKeys;
+
+    	
+       	if (tag.hasAttributes()) {
+    		
+    		mapKeys=tag.attributes();
+    		
+    		if (tag.name().equals("Record")) {
+    			for (Attribute e : mapKeys) {
```

Comment:
This piece of code is rather long, with many if statements (cicloComplexity high), consider refactoring.

can you create an Interface like AttributeValidator, and create as many Validators as required. Then create a Map<String,AttributeValidator> which you can use to find the proper validator instead of so many if statements...

Reply:
For every AttributeValidator we need a unitTest exercising the logic, with proper Assertions on when attribute values are right or wrong.

Reply:
Since valdiations are of type RegExprPattern, legnth range, and boolean type(this should be regularExpr again), maybe create validators at attribute level, instead of verb level, and create an ConfigurableValidator, which perform the same logic with differetn patterns,length range..., so logic is greatly reduced, and the way the validation error is reported unified...